MantidQt/CustomInterfaces/src/EnggDiffraction/EnggDiffractionViewQtGUI.cpp

```diff
@@ -1536,11 +1523,30 @@ void EnggDiffractionViewQtGUI::setDefaultBank(
     if (combo_data > -1) {
       setBankIdComboBox(combo_data);
     } else {
-      setfittingRunNo(selectedFile);
+      setFittingRunNo(selectedFile);
+    }
+  }
+  // check if the vector is not empty so that the first directory
+  // can be assigned to text-field when number is given
+  else if (!m_fitting_runno_dir_vec.empty()) {
+    auto firstDir = m_fitting_runno_dir_vec.at(0);
+    auto intialDir = QString::fromStdString(firstDir);
+    setFittingRunNo(intialDir);
+  }
+  // if nothing found related to text-field input
+  else if (!getFittingRunNo().empty())
+    setFittingRunNo(selectedFile);
+}
+
+bool MantidQt::CustomInterfaces::EnggDiffractionViewQtGUI::isDigit(
```

Comment:
I am using it else where in view too, otherwise I would have to call presenter function from view